// generate via `$ protoc messages.proto  --python_out=./`

syntax = "proto3";
package ga4gh;

message HGVSInfoRequest { }
message HGVSInfoResponse {
  string package_version = 1;
  string rest_api_version  = 2;
  string eval_version  = 3;
  string timestamp  = 4;
  string nomenclature_version  = 5;
}


message Pos {
  int64 start = 1;
  int64 end = 2;
}
message HGVSProjectionResponse {
  string hgvs_string = 1;
}

message HGVSParseRequest {
  string hgvs_string = 1;
}

message HGVSParseResponse {
  string ac  = 1;
  Pos pos = 2;
  string alt = 3 ;
}

message HGVSProjectionRequest {
  string hgvs_string = 1;
  string ac = 2;
}

service HGVSProjectionService {
  rpc GetHGVSInfo(HGVSInfoRequest)
      returns (HGVSInfoResponse) ;
  rpc GetHGVSValidation(HGVSProjectionRequest)
      returns (HGVSProjectionResponse);
  rpc GetHGVSParse(HGVSParseRequest)
      returns (HGVSParseResponse);
  rpc GetHGVSRewrite(HGVSProjectionRequest)
      returns (HGVSProjectionResponse);
}
